corehq/ex-submodules/casexml/apps/case/dbaccessors/related.py

```diff
@@ -30,19 +31,14 @@ def get_reverse_indexed_case_ids(domain, case_ids):
     return [r.case_id for r in get_all_reverse_indices_info(domain, case_ids, CASE_INDEX_CHILD)]
 
 
-def get_reverse_indexed_cases(domain, case_ids, relationship=CASE_INDEX_CHILD):
+def get_reverse_indexed_cases(domain, case_ids, relationship=None):
     """
-    Given a base list of cases, gets all wrapped cases that directly
-    reference them (with relationship <relationship>).
+    Gets all reverse indexed cases of a case (including child cases and extensions).
     """
     from casexml.apps.case.models import CommCareCase
-    keys = [[domain, case_id, 'reverse_index', relationship] for case_id in case_ids]
-    return CommCareCase.view(
-        'case_indices/related',
-        keys=keys,
-        reduce=False,
-        include_docs=True,
-    )
+    indices = get_all_reverse_indices_info(domain, case_ids, relationship)
```

Comment:
What's the rationale behind doing it this way, instead of directly wrapping the output of the `case_indices/related` view with `include_docs=True`? Doesn't this need to make 2 trips to the DB? Or is it so that if there are many ids, the results are chunked?